src/Shared/NativeMethodsShared.cs

```diff
@@ -1310,14 +1310,11 @@ internal static int GetParentProcessId(int processId)
                         if (possibleChildProcess.StartTime > parentStartTime)
                         {
                             int childParentProcessId = GetParentProcessId(possibleChildProcess.Id);
-                            if (childParentProcessId != 0)
+                            if (parentProcessId == childParentProcessId)
```

Comment:
This potentially exposes a path that didn't happen previously (`parentProcessId == 0 && childParentProcessId == 0). What are the chances of this happening? What would even happen if it did?

Reply:
I _believe_ that should never happen. PID 0 is reserved for child processes, that is, the only way to have a pid of 0 is to fork another process, at which point one of the two processes that executes the next code has a pid of 0 to indicate child and the other has a real pid. That means we should never have to worry about them both being 0 unless I'm misunderstanding this, since it's subtle, and I'm slightly hazy.